lib/src/inherited_provider.dart

```diff
@@ -667,37 +667,37 @@ class _CreateInheritedProviderState<T>
             debugIsInInheritedProviderUpdate = true;
             return true;
           }());
-          _value = delegate.update(element, _value);
+          _value = delegate.update!(element!, _value as T);
         } finally {
           assert(() {
             debugIsInInheritedProviderCreate =
-                _debugPreviousIsInInheritedProviderCreate;
+                _debugPreviousIsInInheritedProviderCreate!;
             debugIsInInheritedProviderUpdate =
-                _debugPreviousIsInInheritedProviderUpdate;
+                _debugPreviousIsInInheritedProviderUpdate!;
             return true;
           }());
         }
 
         assert(() {
-          delegate.debugCheckInvalidValueType?.call(_value);
+          delegate.debugCheckInvalidValueType?.call(_value as T);
           return true;
         }());
       }
     }
 
-    element._isNotifyDependentsEnabled = false;
-    _removeListener ??= delegate.startListening?.call(element, _value);
-    element._isNotifyDependentsEnabled = true;
+    element!._isNotifyDependentsEnabled = false;
+    _removeListener ??= delegate.startListening?.call(element!, _value as T);
+    element!._isNotifyDependentsEnabled = true;
     assert(delegate.startListening == null || _removeListener != null);
-    return _value;
+    return _value as T;
```

Comment:
I think this is incorrect. Previously if a value was null we would throw a `ProviderNotFoundException` however, with this change I believe we will instead have a `_CastError` type 'Null' is not a subtype of type 'T' in type cast.
```suggestion
    return _value as T?;
```

Reply:
We cannot return null. That would make context.watch return T?

Reply:
Thoughts on something like:

```dart
if (_value == null) throw ProviderNotFound(...);
return _value as T;
```

I don't think the `CastError` should be propagated to developers. It should be handled internally and surfaced as the previous `ProviderNotFound` exception. Thoughts?

Reply:
Null may be a valid value

To begin with, "create" should not throw.
It may need a custom error ( i don't believe ProviderNotFound fits here), but it definitely should throw

Reply:
Yeah good point.

As I mentioned above, I think it's valid to perform a read from within a create which could potentially fail:

```dart
Provider<String>(
  create: (context) {
    return '${context.read<int>()}';
  },
  child: ...
);
```

If the upstream provider is not found, `create` will throw and I'm fine with a custom error (maybe `ProviderCreateException`?).

The main things imo are:

- The case described above should be able to be handled by other providers
- The error shown to the developer should be easy to understand and point to the root cause

Reply:
The changes in https://github.com/rrousselGit/provider/pull/574/commits/4838869adc2be94e02c903b4057b76d067bad61f address this issue and make it backward compatible with the previous behavior. @rrousselGit can you please take a look and let me know what you think? I verified that the flutter_bloc test suite now passes, thanks! 🙏